From 60b0ab5c4f7f15454e91902b5ecd3e629a930b32 Mon Sep 17 00:00:00 2001
From: Yu Watanabe <watanabe.yu+github@gmail.com>
Date: Wed, 23 Aug 2017 12:36:36 +0900
Subject: [PATCH] networkd: do not fail manager_connect_bus() if dbus is not
 active yet

Fixes #6618.

(cherry picked from commit fb72b1d99f661ea62fd534e4bc1174c6337611c8)
---
 src/network/networkd-manager.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/network/networkd-manager.c b/src/network/networkd-manager.c
index 5f10b4f993..718c8bb03f 100644
--- a/src/network/networkd-manager.c
+++ b/src/network/networkd-manager.c
@@ -136,7 +136,7 @@ int manager_connect_bus(Manager *m) {
         assert(m);
 
         r = sd_bus_default_system(&m->bus);
-        if (r == -ENOENT) {
+        if (r < 0) {
                 /* We failed to connect? Yuck, we must be in early
                  * boot. Let's try in 5s again. As soon as we have
                  * kdbus we can stop doing this... */
